module/entities/helper.js

```diff
@@ -595,4 +596,110 @@ module.exports = class EntitiesHelper {
 
 }
 
+    /**
+   * Get users by entityId and role
+   * @method
+   * @name getUsersByEntityAndRole
+   * @param {Object} requestedData - requested data.
+   * @param {String} req.body.entityId - entity id.
+   * @param {String} req.body.role - role code.
+   * @returns {Array}  - List of userIds and entityIds
+   */
+
+  static getUsersByEntityAndRole( entityId= "", role= "" ) {
+    return new Promise(async (resolve, reject) => {
+        try {
+
+            if (entityId == "") {
+                throw new Error(constants.apiResponses.ENTITY_ID_REQUIRED);
+            }
+
+            if (role == "") {
+                throw new Error(constants.apiResponses.ROLE_REQUIRED)
+            }
+            
+            let entityDocument = await this.entityDocuments
+            (
+                {
+                    _id: entityId
+                },
+                [
+                    "entityType" 
+                ]
+            )
+            
+            if (!entityDocument.length) {
+                throw new Error(constants.apiResponses.INVALID_ENTITY_ID);
+            }
+
+            let userRoles = await userRolesHelper.roleDocuments({
+                code : role
+            },[
+                "entityTypes"
+            ]);
+            
+            if(!userRoles.length) {
+                throw new Error(constants.apiResponses.INVALID_ROLE)
+            }
+
+            let queryObject = {
```

Comment:
@deepakr123 - Let's change as per our discussion.

Fetch all sub-entities from MongoDB
For sub-entities get the doc from ES with a projection of only the required role.

Reply:
resolved